### PR TITLE
[RNMobile][Monorepo] Make the native app customizable by a parent app

### PR DESCRIPTION
**Pull request by Tug**
## Description
This PR does multiple changes which allows external native apps to reuse and customize the native block editor.
The main repo which customizes WordPress apps for Android and iOS is still gutenberg-mobile.
Here is the related PR for this integration: https://github.com/wordpress-mobile/gutenberg-mobile/pull/2172

List of changes:
- Create a `render` method in react platform to mimick the web API. Update the code in `edit-post` to use this new method
- Run 2 hooks in the `render` method to customize root component props and run pre render initialization code
- Fix `packages/react-native-bridge/android/build.gradle` to make WPAndroid build (switch from yarn to npm, update paths...)
- Commit hermes android library into `packages/react-native-bridge/android/vendor/hermes-engine` so we can build WPAndroid using jitpack and a jsbundle without running `npm install`
- Fix e2e tests in gutenberg-mobile

## How has this been tested?
See Gutenberg mobile PR

## Types of changes
React native changes

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style. 
- [ ] My code follows the accessibility standards. 
- [ ] My code has proper inline documentation. 
- [ ] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/react-native-editor/ios/Podfile [hunk 1]
```diff
@@ -43,6 +43,8 @@ target 'gutenberg' do
   pod 'Gutenberg', :path => '../../react-native-bridge/Gutenberg.podspec'
 
 
+  pod 'react-native-get-random-values', :path => '../node_modules/react-native-get-random-values'
```
Comment (ceyhun) [inline packages/react-native-editor/ios/Podfile:46]: No need to add this to Podfile, it should be "autolinked" :)

File: packages/element/src/react-platform.native.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+import { applyFilters, doAction } from '@wordpress/hooks';
+
+/**
+ * Internal dependencies
+ */
+import { cloneElement } from './react';
+
+const render = ( element, id ) =>
+	AppRegistry.registerComponent( id, () => ( propsFromParent ) => {
```
Comment (gziolo) [inline packages/element/src/react-platform.native.js:18]: Out of curiosity, why not to execute all this logic inside initializeEditor in the @wordpress/edit-post package?

Comment (ceyhun) [reply packages/element/src/react-platform.native.js:18]: I think it's to be able to customize the editor from an outside package. In this case a repo that's importing Gutenberg as a submodule. Correct me if I'm wrong @Tug?

Comment (Tug) [reply packages/element/src/react-platform.native.js:18]: No we could totally have this logic in `@wordpress/edit-post` (or in `@wordpress/react-native-editor` as it was the case before in the monorepo branch before this PR).

The idea here is simply to get closer to having `@wordpress/edit-post` work as cross-platform package. If you look at [`edit-post/src/index.native.js`](https://github.com/WordPress/gutenberg/blob/66c0dc7c709fd0a3fe7a794ca56e304eee5abc8c/packages/edit-post/src/index.native.js`) 
you'll see that the init code is much closer to the [web one](https://github.com/WordPress/gutenberg/blob/66c0dc7c709fd0a3fe7a794ca56e304eee5abc8c/packages/edit-post/src/index.js#L138) now.

> Overriding React methods and adding WP hooks might have some performance implications but I guess it’s called only once.

We're not overriding anything though, the `render` function wasn't defined for native prior to this. And yes it's only rendered once on load.

Comment (gziolo) [reply packages/element/src/react-platform.native.js:18]: It probably doesn't matter as much then now, but I guess `initializeEditor` would be the best place to run all those side effects.

**Comment by gziolo** [comment]
@ceyhun and @Tug – I don't see any blocker from the web side of things. It's great to see progress on integration 👍

**Review by cameronvoell — APPROVED** [review]
Ready to merge, using the usual “Create a merge commit” option for preserving history on our `feat/import-gutenberg-mobile-no-squash` branch

**Comment by hypest** [comment]
Will now merge with "Use merge commit" strategy and revert back to "squash/rebase strategy" right after.

**Pull request merged**